bat/ssh.go

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package bat
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"golang.org/x/crypto/ssh"
+)
+
+//SSHClient containes details of the server to be connect
+type SSHClient struct {
+	ClientIP   string
+	ClientPort string
+	ClientUser string
+	ClientPass string
+}
+
+//CreateSSHClient creates a SSH session for the server specified
+//
+//Receive a struct with the details of the server that will be connected
+//crete a tcp session that will be use to send and receive data
+//
+//sesion sould be created succesfully and wait for commands
+func CreateSSHClient(WLclient SSHClient) (*ssh.Client, error) {
```

Comment:
We need the extra dial function I'm afraid.  In fact I'm surprised it doesn't already exist in the ssh package.   The reason is that each BAT test creates a context at the start of the test and they use this to time out all the calls they make.  This lets us set a timeout per test rather than per call.  The function should be easy to write though.

```
func DialSSH(network, addr string, config *ClientConfig, ctx context.Context) (*ssh.Client, error) {
	conn, err := net.DialContext(ctx, network, addr)
	if err != nil {
		return nil, fmt.Errorf("Failed to connect to ssh server: %v", err)
	}
	c, chs, r, err := ssh.NewClientConn(conn, addr, config)
	if err != nil {
		return nil, fmt.Errorf("Failed to create new SSH client connection : %v", err)
	}
	return ssh.NewClient(c, chs, r), nil
}
```

or something like that.